### PR TITLE
Add fallback component for handling errors and additional error throws

### PR DESCRIPTION
**Pull request by mgramigna**
Restructured the way that our errors are handled. You can see the various cases that throw errors now by inspecting the code.

For testing, I'd recommend ensuring that one of these cases happen (e.g. modify the patient bundle or the questionnaire to be invalid), and see what happens in the UI. It should display an error as well as the stack trace.

**Important Note:** When running the react app in development mode, an overlay will pop up when the error is thrown. This will not be viewable in production, so to see the actual error component, you need to close out of this by hitting the X in the top right.

### CODE REVIEW

**Comment by pmadria** [comment]
Tested it out and the errors pop up with the alert dialog...would we want to add error handling to make sure that the language is text/cql and type is of a valid type? Not sure if that's too specific for the task.

File: src/App.tsx [hunk 1]
```diff
@@ -25,41 +26,55 @@ const App = () => {
...
+        e => e.url === 'http://hl7.org/fhir/StructureDefinition/cqf-library'
+      );
 
-        if (extension && extension.valueCanonical) {
-          const response = await axios.get(extension.valueCanonical);
-          const fhirLibrary = response.data as R4.ILibrary;
-          const library = await new LibraryLoader(fhirLibrary).fetchELM();
+      if (!extension || !extension.valueCanonical) {
+        throw new Error(
+          'Provided questionnaire does not contain a proper extension for "http://hl7.org/fhir/StructureDefinition/cqf-library"'
```
Comment (mickohanlon23) [inline src/App.tsx:39]: When I trigger this error by modifying the questionnaire extension, it comes up in development mode overlay as an Unhandled Rejection, and then when I "X" out of the overlay it doesn't show the error component, it is just stuck on the "Loading..." message. Do you know why this is? In comparison, the error component did show properly when I tweaked the patient bundle.

File: src/App.tsx [hunk 2]
```diff
@@ -25,41 +26,55 @@ const App = () => {
...
+      const response = await axios.get(extension.valueCanonical);
+      const fhirLibrary = response.data as R4.ILibrary;
 
-          setLibrary(library);
-          setValueSetMap(valueSetMap);
-        }
-      } catch (e) {
-        console.error(`Error loading questionnaire data: ${e.message}`);
+      if (!R4.RTTI_Library.decode(fhirLibrary).isRight()) {
+        throw new Error(`${extension.valueCanonical} did not provide a valid FHIR library`);
```
Comment (mickohanlon23) [inline src/App.tsx:47]: Same issue as above when I modify the library

**Review by mickohanlon23 — CHANGES_REQUESTED** [review]
After rebasing to master, the error component rendered when I triggered the errors in `Abstraction.tsx` and `results-processing.ts`, but when the error throws in `App.tsx` were triggered, no component displayed.

**Comment by mgramigna** [comment]
Ok I think I fixed everything. More changes than expected, so lmk if you have questions

**Review by mickohanlon23 — COMMENTED** [review]
I tested all of the error conditions. The errors in `Abstractor.tsx` are rendering as expected. The error components in `App.tsx` are rendering, but are not showing a stack trace. And when I generate errors in `QuestionnaireLoader.ts` and `ValueSetLoader.ts`, the error message is not rendered at all after closing out the overlay that exists in development mode. This might not be too critical, since we're only running in development mode for now anyways.

Other than these issues, I finished looking through everything, and the new code structure looks good. Reformatting in `Abstractor.tsx` looks great too.

**Pull request merged**